src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -325,12 +335,16 @@ void WindowMoveHandlerPrivate::MoveSizeEnd(HWND window, POINT const& ptScreen, c
         auto zoneWindow = std::move(m_zoneWindowMoveSize);
         ResetWindowTransparency();
 
-        auto windowInfo = FancyZonesUtils::GetFancyZonesWindowInfo(window);
+        bool hasNoVisibleOwnoer = FancyZonesUtils::HasNoVisibleOwner(window);
+        bool isStandardWindow = FancyZonesUtils::IsStandardWindow(window);
+        auto windowState = get_window_state(window);
 
-        if (windowInfo.standardWindow == false && windowInfo.noVisibleOwner == false &&
-          m_moveSizeStartWindowInfo.standardWindow == true && m_moveSizeStartWindowInfo.noVisibleOwner == true)
+        if ((isStandardWindow == false && hasNoVisibleOwnoer == false &&
+             m_moveSizeWindowInfo.standardWindow == true && m_moveSizeWindowInfo.noVisibleOwner == true) ||
+            (windowState == WindowState::MAXIMIZED))

```

Comment:
We have utility function `IsWindowMaximized` inside FancyZones (check `util.h`). Maybe it's better to use that one as it is performing much simpler logic.

Reply:
Nice. Fixed